kernel/fatal.c

```diff
@@ -117,5 +119,30 @@ void z_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
 		__ASSERT(!k_is_in_isr(),
 			 "Attempted to recover from a fatal error in ISR");
 	}
+
+#if defined(z_arch_is_in_nested_exception)
```

Comment:
I don't think this isn't the best way to do this.
we need to put a proper prototype in include/sys/arch_interface.h.
And either provide an implementation for every arch, or file a GH issue and introduce some short-lived Kconfig CONFIG_ARCH_HAS_IS_IN_NESTED_EXCEPTION or something like that until we can get parity on all arches.

Reply:
Fine, with me. Since your work is now merged, i can introduce this short-lived Kconfig option, and select it for ARM architecture, and then add the prototype in arch_interface.h

Reply:
Done this way.